dwave/cloud/solver.py

```diff
@@ -345,49 +377,108 @@ def _encode_problem_as_ref(self, problem, params):
         else:
             logger.debug("To encode the problem for submit in the 'ref' format, "
                          "we need to upload it first.")
-            problem_id = self.upload_bqm(problem).result()
+            problem_id = self.upload_problem(problem).result()
 
         body = json.dumps({
             'solver': self.id,
             'data': encode_problem_as_ref(problem_id),
-            'type': 'bqm',
+            'type': problem_type,
             'params': params
         })
         logger.trace("Sampling request encoded as: %s", body)
 
         return body
 
-    def sample_bqm(self, bqm, **params):
-        """Sample from the specified :term:`BQM`.
+    def sample_problem(self, problem, problem_type=None, **params):
+        """Sample from the specified problem.
 
         Args:
-            bqm (:class:`~dimod.BinaryQuadraticModel`/str):
-                A binary quadratic model, or a reference to one
-                (Problem ID returned by `.upload_bqm` method).
+            problem (dimod-model-like/str):
+                A quadratic model (e.g. :class:`dimod.BQM`/:class:`dimod.DQM`),
+                or a reference to one (Problem ID returned by
+                :meth:`.upload_problem` method).
+
+            problem_type (str, optional):
+                Problem type, one of the handled problem types by the solver.
+                If not specified, the first handled problem type is used.
 
             **params:
                 Parameters for the sampling method, solver-specific.
 
         Returns:
             :class:`~dwave.cloud.computation.Future`
-
-        Note:
-            To use this method, dimod package has to be installed.
         """
 
+        # infer problem_type; for now just the the first handled (always just one)
+        if problem_type is None:
+            problem_type = next(iter(self._handled_problem_types))
+
         # encode the request (body as future)
         body = self.client._encode_problem_executor.submit(
-            self._encode_problem_as_ref,
-            problem=bqm, params=params)
+            self._encode_problem_for_submission,
+            problem=problem, problem_type=problem_type, params=params)
 
         # computation future holds a reference to the remote job
-        computation = Future(solver=self, id_=None, return_matrix=self.return_matrix)
+        computation = Future(
+            solver=self, id_=None, return_matrix=self.return_matrix)
 
         logger.debug("Submitting new problem to: %s", self.id)
         self.client._submit(body, computation)
 
         return computation
 
+
+class BQMSolver(BaseUnstructuredSolver):
+    """Class for D-Wave unstructured binary quadratic model solvers.
```

Comment:
"Class for D-Wave unstructured" --> I think we generally refer to these as Leap rather than D-Wave